test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +27,130 @@ describe('Challenge Generation', (): void => {
     expect(Buffer.byteLength(challenge, 'base64')).toBe(8)
   })
 })
+
+describe('Signature Verification', (): void => {
+  // Repeat test with different keys
+  for (let index = 0; index < 4; index++) {
+    it('verifies correct signature - EC Key', async (): Promise<void> => {
+      const keyPair = crypto.generateKeyPairSync('ec', {
+        namedCurve: 'secp256k1', // Allowed by FIDO spec
+        publicKeyEncoding: {
+          type: 'spki', // Key infrasructure
+          format: 'pem' // Encoding format
+        },
+        privateKeyEncoding: {
+          type: 'pkcs8',
+          format: 'pem'
+        }
+      })
+
+      const challenge = 'Crypto Auth service Yay!'
+      const signer = crypto.createSign('SHA256')
+
+      signer.update(challenge)
+
+      const sign = signer.sign(keyPair.privateKey, 'base64')
+
+      expect(await verifySign(challenge, sign, keyPair.publicKey)).toEqual(true)
+    })
+  }
+
+  // Repeat test with different keys
+  for (let index = 0; index < 4; index++) {
+    it('returns false on incorrect signature - wrong EC key', async (): Promise<void> => {
+      const realKeyPair = crypto.generateKeyPairSync('ec', {
+        namedCurve: 'secp256k1', // Allowed by FIDO spec
+        publicKeyEncoding: {
+          type: 'spki', // Key infrasructure
+          format: 'pem' // Encoding format
+        },
+        privateKeyEncoding: {
+          type: 'pkcs8',
+          format: 'pem'
+        }
+      })
+
+      const fakeKeyPair = crypto.generateKeyPairSync('ec', {
+        namedCurve: 'secp256k1', // Allowed by FIDO spec
+        publicKeyEncoding: {
+          type: 'spki', // Key infrasructure
+          format: 'pem' // Encoding format
+        },
+        privateKeyEncoding: {
+          type: 'pkcs8',
+          format: 'pem'
+        }
+      })
+
+      const challenge = 'Crypto Auth service Yay!'
+      const signer = crypto.createSign('SHA256')
+
+      signer.update(challenge)
+
+      const sign = signer.sign(fakeKeyPair.privateKey, 'base64')
+
+      expect(await verifySign(challenge, sign, realKeyPair.publicKey)).toEqual(false)
+    })
+  }
+
+  it('verifies correct signature - RSA Key', async (): Promise<void> => {
+    const realKeyPair = crypto.generateKeyPairSync('rsa', {
+      modulusLength: 2048 // Key length in bits
+    })
+
+    const challenge = 'Crypto Auth service Yay!'
```

Comment:
but also if you're using the same challenge + calling the same functions everytime - consider using beforeAll() beforeEach() in jest to improve code readability